src/components/SignInButton.tsx

```diff
@@ -0,0 +1,212 @@
+import React, { useState } from 'react';
+import { isEmpty, isNil } from 'lodash/fp';
+import { Alert } from './Alert';
+import { ToS } from '../libs/ajax/ToS';
+import { DuosUser, User } from '../libs/ajax/User';
+import { Metrics } from '../libs/ajax/Metrics';
+import { Storage } from '../libs/storage';
+import { Navigation, setUserRoleStatuses } from '../libs/utils';
+import loadingImage from '../images/loading-indicator.svg';
+import { Auth } from '../libs/auth/auth';
+import eventList from '../libs/events';
+import { StackdriverReporter } from '../libs/stackdriverReporter';
+import CSS from 'csstype';
+import { useHistory } from 'react-router';
+import { OidcUser } from 'src/libs/auth/oidcBroker';
+
+interface SignInButtonProps {
+  customStyle: CSS.Properties | undefined;
```

Comment:
Nit: suggest naming this prop `style` so it's more intuitive.

```suggestion
  style: CSS.Properties | undefined;
```